src/Query/Scanner.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Removes newlines escaped by a backslash.
+ * A trailing backslash at the end of a line can be escaped by doubling it.
+ *
+ * @param input input string
+ * @returns modified input
+ */
+export function continue_lines(input: string): string {
+    return input.replace(/(?<!\\)\\\n/g, '').replace(/\\\\\n/g, '\\\n');
```

Comment:
That's unfortunate! This feature allows a final backslash to be escaped, eg

    ```tasks
    description includes \\
    explain
    ```

Without lookbehind this is harder to describe in regex.

Maybe I should drop that feature? Whitespace can still be used at the end of the line after a backslash (Obsidian seems to preserve this, but it some formatting tools strip trailing spaces). One could also use a mustache comment after the backslash, although I don't think that feature is widely known.

    ```tasks
    description includes \{{!}}
    explain
    ```

Reply:
Lookbehind is removed in the latest commit. It slightly changes the double-backslash escape in that you also need an extra newline to eat the final backslash:

    ```tasks
    description includes \\
    
    explain
    ```

> description includes \

This is a super rare case, so I think I'm probably bringing it up too much.

Reply:
_Picking one of a few possible locations to comment on handling of whitespace at either end of lines..._

I've thought about this a lot since reading your replies, and currently my feeling is that the guiding principle should be that Tasks has always stripped whitespace from both ends of each line in the query, like this:

```ts
    constructor(source: string, path: string | undefined = undefined) {
        this.source = source;
        this.filePath = path;

        source
            .split('\n')
            .map((rawLine: string) => rawLine.trim())
            .forEach((rawLine: string) => {
```

I think the more knowledgeable users will expect this - and the less knowledgable ones will likely accept whatever statement we make in the docs about how leading and trailing spaces are handled.

So, my feeling is that I would prefer this feature to consistently make all of these:

```tasks
explain

description includes hello world

description includes hello\
world

description includes hello \
world

description includes hello    \
world
```

become...

```
description includes hello world
```

Reasons include, in decreasing order of importance:

- It is less likely for searches to fail to find tasks that the user thought would match their query
- It is less surprising for users... Invisible space in their query will not silently change the search behaviour (since most people don't `explain` all the time)
- It is easier to document (we don't have to explain the different behaviours...)
- Smaller point: it will lead to more readable `explain` output in heavily indented custom filters and groups
- Fewer test cases to write, as varieties of behaviour need to be tested

Reply:
And so replying to your actual question, I am comfortable with removing the ability to escape the final backslash.

We can say in the documentation what the behavior is in that case, and show how to write the equivalent query with `filter by function`.